--- a/luci-app-adguardhome/root/etc/init.d/AdGuardHome
+++ b/luci-app-adguardhome/root/etc/init.d/AdGuardHome
@@ -29,7 +29,7 @@ set_forward_dnsmasq()
 		if [ "$server" = "$addr" ]; then
 			continue
 		fi
-		uci add_list dhcp.@dnsmasq[0].server=$server
+		# uci add_list dhcp.@dnsmasq[0].server=$server
 	done
 	uci delete dhcp.@dnsmasq[0].resolvfile 2>/dev/null
 	uci set dhcp.@dnsmasq[0].noresolv=1
@@ -499,6 +499,14 @@ start_service() {
 	echo "AdGuardHome service enabled"
 	echo "luci enable switch=$enabled"
 	(sleep 10 && [ -z "$(pgrep $binpath)" ] && logger "AdGuardHome" "no process in 10s cancel redirect" && _do_redirect 0 )&
+	if [[ "`uci get bypass.@global[0].adguardhome 2>/dev/null`" == 1 ]]; then
+		c1=0
+		while ! ps -w | grep -i AdGuardHome | grep -v grep >/dev/null 2>&1; do
+			sleep 1
+			[ $c1 == 5 ] && break || let c1++
+		done
+		/etc/init.d/bypass reload 2>/dev/null
+	fi
 }
 
 reload_service()
