--- a/luci-app-adguardhome/root/etc/init.d/AdGuardHome
+++ b/luci-app-adguardhome/root/etc/init.d/AdGuardHome
@@ -29,7 +29,7 @@ set_forward_dnsmasq()
 		if [ "$server" = "$addr" ]; then
 			continue
 		fi
-		uci add_list dhcp.@dnsmasq[0].server=$server
+		# uci add_list dhcp.@dnsmasq[0].server=$server
 	done
 	uci delete dhcp.@dnsmasq[0].resolvfile 2>/dev/null
 	uci set dhcp.@dnsmasq[0].noresolv=1
@@ -499,6 +499,13 @@ start_service() {
 	echo "AdGuardHome service enabled"
 	echo "luci enable switch=$enabled"
 	(sleep 10 && [ -z "$(pgrep $binpath)" ] && logger "AdGuardHome" "no process in 10s cancel redirect" && _do_redirect 0 )&
+	if [[ "`uci get bypass.@global[0].adguardhome 2>/dev/null`" == 1 ]]; then
+		c1=0
+		while ! grep -i 5553 /var/dnsmasq.d/dnsmasq-by.conf 2>&1; do
+		/etc/init.d/bypass reload 2>/dev/null
+			[ $c1 == 5 ] && break || let c1++
+		done
+	fi
 }
 
 reload_service()

--- a/adguardhome/Makefile
+++ b/adguardhome/Makefile
@@ -52,6 +52,11 @@ define Package/adguardhome/conffiles
 /etc/adguardhome.yaml
 endef
 
+define Package/adguardhome/postinst
+#!/bin/sh
+	/etc/init.d/AdGuardHome restart
+endef
+
 define Package/adguardhome/description
 Free and open source, powerful network-wide ads and trackers blocking DNS server.
 endef
