--- a/luci-app-ssr-plus/luasrc/model/cbi/shadowsocksr/servers.lua
+++ b/luci-app-ssr-plus/luasrc/model/cbi/shadowsocksr/servers.lua
@@ -41,5 +41,5 @@
 o.rmempty=false
 o.description=translate("Subscribe new add server default Auto-Switch on")
-o.default=1
+o.default=0
 
 o=s:option(Flag,"proxy",translate("Through proxy update"))

--- a/luci-app-ssr-plus/Makefile
+++ b/luci-app-ssr-plus/Makefile
@@ -102,4 +102,5 @@
 
 define Package/$(PKG_NAME)/conffiles
+/etc/config/shadowsocksr
 /etc/ssr/
 endef
@@ -133,4 +134,5 @@
 #!/bin/sh
 if [ -z "$${IPKG_INSTROOT}" ]; then
+	/etc/init.d/shadowsocksr enable
 	/etc/uci-defaults/luci-ssr-plus && rm -f /etc/uci-defaults/luci-ssr-plus
 fi

--- a/luci-app-ssr-plus/root/etc/init.d/shadowsocksr
+++ b/luci-app-ssr-plus/root/etc/init.d/shadowsocksr
@@ -657,5 +657,5 @@
 
 start_switch(){
-	if [ $(uci_get_by_type global enable_switch 0) = 1 -a $(uci_get_by_name $GLOBAL_SERVER switch_enable 0) = 1 -a -z "$switch_server" ];then
+	if [ $(uci_get_by_type global enable_switch 0) = 1 -a -z "$switch_server" ];then
 		service_start $BIN_DIR/ssr-switch start
 		switch_enable=1
